Update the release note for language changes (#1)
praneesha/release-notes: release-notes-1.2.0.md

- `lang.array` module - New functions added: `lastIndexOf()`, `toStream()`
- `lang.map` module - New functions added: `toArray()`, `removeIfHasKey()`

### Backward incompatible improvements and bug fixes
- Listener variables are now final.
- The syntax for Unicode escapes in strings has changed from `\u[CodePoint]` to `\u{CodePoint}` so as to align with ECMAScript. Although this is an incompatible change, the previous syntax was not implemented.

> - The semantics of the `lock` statement has been changed. Now it acquires a single, program-wide, recursive mutex before executing a lock statement, and releases the mutex after completing the execution of the lock statement.

Reviewer: ```suggestion
- The semantics of the `lock` statement has been changed. Now, it acquires a single, program-wide, recursive mutex before executing a lock statement and releases the mutex after completing the execution of the lock statement.
```

> - When a list constructor or a mapping constructor is used without a contextually expected type, we now infer a tuple or a record type rather than an array or a map type.

Reviewer: ```suggestion
- When a list constructor or a mapping constructor is used without a contextually-expected type, now a tuple or a record type is inferred rather than an array or a map type.
```

> - The syntax `x@` for accessing the attributes of an XML element has been removed.

Reviewer: ```suggestion
- The `x@` syntax  for accessing the attributes of an XML element has been removed.
```

> - lvalue accesses of dynamic lists does a filling read.

Reviewer 2: Term "lvalue access" is too technical. Maybe we can say, "Now member value assignment of a dynamic list adds filter values if required members are not already filled."

Author: Yeah, will change.

```suggestion
- Member value assignment for a dynamic list now adds filler values if the required members are not already filled and the relevant member type has a filler value.
```

> - lvalue accesses of dynamic lists does a filling read.

Reviewer 3: We haven't introduced lvalues anywhere. It's only in the spec. So shall we use a simpler way to say it? I remember @MaryamZi once did so in a BBE I think

> - Member value assignment for a dynamic list now adds filler values if the required members are not already filled and the relevant member type has a filler value.

Reviewer: ```suggestion
- Now, member value assignment for a dynamic list adds filler values if the required members are not already filled and the relevant member type has a filler value.
```

> - Functions with the same name were previously allowed in both the module and the module's tests. Now the same gives a compilation error.

Reviewer 3: ```suggestion
- Functions with the same name were previously allowed in both the module and the module's tests. This is now disallowed and results in a compilation error.
```

> - Out of index access of `xml` sequences now result in empty sequences rather than a panic. To achieve panic behavior the `xml:get()` langlib function can be used.

Reviewer 3: The second sentence sounds a bit odd. Is there a better way to phrase it?

Author: How about
```suggestion
- Out of range index access of `xml` sequences now result in empty sequences rather than a panic. Using the `xml:get()` langlib method with an out of range index results in a panic.
```

> - Out of range index access of `xml` sequences now result in empty sequences rather than a panic. Using the `xml:get()` langlib method with an out of range index results in a panic.

Reviewer: ```suggestion
- Now, out of range index access of `xml` sequences result in empty sequences rather than a panic. Using the `xml:get()` langlib method with an out of range index results in a panic.
```
